File: advanced-programming-methods/labs/toy-interpreter-java/src/model/IList.java
package model;

/**
 * The interface I list.
 */
public interface IList extends java.io.Serializable
{
    /**
     * Add void.
     *
     * @param value the value
     */
    public void add(String value);

    /**
     * Gets last.
     *
     * @return the last
     */
    public String getLast();

    /**
     * Gets all.
     *
     * @return the all
     */
    public String getAll();

    /**
     * Clear void.
     */
    public void clear();
}

File: advanced-programming-methods/labs/toy-interpreter-sharp/model/IList.cs
using System;

namespace Model
{
    /// <summary>
    /// The interface I list.
    /// </summary>
    public interface IList
    {
        /// <summary>
        /// Add void.
        /// </summary>
        /// <param name="value"> the value </param>
        void Add(string value);

        /// <summary>
        /// Gets last.
        /// </summary>
        /// <returns> the last </returns>
        string Last {get;}

        /// <summary>
        /// Gets all.
        /// </summary>
        /// <returns> the all </returns>
        string All {get;}

        /// <summary>
        /// Clear void.
        /// </summary>
        void Clear();
    }
}
